config/extenders/codeSplitting.js

```diff
@@ -0,0 +1,162 @@
+  // registered within the following configuration setting.
+  const whitelist = [
+    /\.bin/,
+    'source-map-support/register',
+    'react-universal-component',
+    'webpack-flush-chunks',
+  ]
+    // And any items that have been whitelisted in the config need
+    // to be included in the bundling process too.
+    .concat(config('nodeExternalsFileTypeWhitelist') || []);
+
+  return fs
+    .readdirSync(path.resolve(appRootDir.get(), 'node_modules'))
+    .filter(x => !whitelist.some((w) => {
+      if (w instanceof RegExp) {
+        return w.test(x);
+      }
+      return x === w;
+    }))
+    .reduce((ext, mod) => {
+      // mark this module as external
+      // https://webpack.js.org/configuration/externals
+      ext[mod] = `commonjs ${mod}`;
+      return ext;
+    }, {});
+}
+
+export default (webpackConfig, buildOptions) => {
+  const { target, optimize = false, localIdentName } = buildOptions;
+
+  const isProd = optimize;
+  const isDev = !isProd;
+  const isClient = target === 'client';
+  const isNode = !isClient;
+
+  const ifNode = ifElse(isNode);
+  const ifClient = ifElse(isClient);
+  const ifDevClient = ifElse(isDev && isClient);
+
+  // Overwrite the externals because apparently `webpack-node-externals` does not
+  // work well with `webpack-flush-chunks`
+  if (isNode) {
+    webpackConfig.externals = externals();
+  }
+
+  // Remove ExtractTextPlugin
+  const etpIndex = webpackConfig.plugins.findIndex(p => p instanceof ExtractTextPlugin);
+  if (etpIndex > -1) {
+    webpackConfig.plugins.splice(etpIndex, 1);
+  }
+  // Add some plugins for css code splitting
+  webpackConfig.plugins.push(
+    ...removeNil([
+      ifDevClient(new WriteFilePlugin()),
+      ifClient(new ExtractCssChunks()),
+      ifClient(
+        new webpack.optimize.CommonsChunkPlugin({
+          names: ['bootstrap'], // needed to put webpack bootstrap code before chunks
+          filename: isDev ? '[name].js' : '[name]-[chunkhash].js',
+          minChunks: Infinity,
+        }),
+      ),
+      ifNode(
+        new webpack.optimize.LimitChunkCountPlugin({
+          maxChunks: 1,
+        }),
+      ),
+    ],
+  ));
+
+  // Overwrite happypack css dev plugin
+  const happyPackDevclientCssIndex = webpackConfig.plugins.findIndex(r => r.id === 'happypack-devclient-css');
+  if (happyPackDevclientCssIndex > 1) {
+    webpackConfig.plugins.splice(
+      happyPackDevclientCssIndex,
+      1,
+      happyPackPlugin({
+        name: 'happypack-devclient-css',
+        loaders: [
+          'classnames-loader',
+          ...ExtractCssChunks.extract({
+            fallback: 'style-loader',
+            use: [
+              `css-loader?sourceMap=1&modules=1&importLoaders=1&localIdentName=${localIdentName}`,
+              'postcss-loader?sourceMap=1',
+              'sass-loader?sourceMap=1&outputStyle=expanded',
+            ],
+          }),
+        ],
+      }),
+    );
+  }
```

Comment:
Maybe we don’t want to do css code splitting during dev?
Then we don’t need this and don’t need `WriteFilePlugin`